### PR TITLE
[libsecret] add gcrypt option

### PR DESCRIPTION
**Pull request by hrantzsch**
Specify library name and version:  **libsecret/0.20.4**

Add support for using gcrypt on Linux. The libgcrypt recipe is unfortunately not available on other platforms at this time.

Note that the option is set to true by default, so this **changes the behavior of the recipe**. This does reflect the original default behavior of libsecret though (and it's safer to use gcrypt).

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libsecret/all/conanfile.py [hunk 1]
```diff
@@ -14,8 +15,8 @@ class LibsecretConan(ConanFile):
     generators = "pkg_config"
 
     settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {"shared": False, "fPIC": True}
+    options = {"shared": [True, False], "fPIC": [True, False], "gcrypt": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "gcrypt": True}
```
Comment (madebr) [inline recipes/libsecret/all/conanfile.py:19]: Please reformat + use `with_libgcrypt`.
That's the format we often use to let the user know that the option will add support for an extra package.
```suggestion
    options = {
        "shared": [True, False],
        "fPIC": [True, False],
        "with_libgcrypt": [True, False],
    }
    default_options = {
        "shared": False,
        "fPIC": True,
        "with_libgcrypt": True,
    }
```

File: recipes/libsecret/all/conanfile.py [hunk 2]
```diff
@@ -34,6 +47,11 @@ def validate(self):
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if self.options.with_libgcrypt and self.settings.os != "Linux":
```
Comment (uilianries) [inline recipes/libsecret/all/conanfile.py:50]: Need to move it to `configure()` method. When `config_options` is called, the `self.options.with_libgcrypt` is not processed yet, so its value will be the default option value always. On the other hand, `configure()` is one step later and all options are processed there.

Comment (hrantzsch) [reply recipes/libsecret/all/conanfile.py:50]: Oh! I misunderstood that. Thanks for the explanation.

**Review by uilianries — DISMISSED** [review]
LGTM

File: recipes/libsecret/all/conanfile.py [hunk 3]
```diff
@@ -38,11 +51,21 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
+
+        if self.options.with_libgcrypt and self.settings.os != "Linux":
+            self.output.warn(
+                "gcrypt is only available on Linux. Option will be ignored."
+            )
+            del self.options.with_libgcrypt
```
Comment (prince-chrismc) [inline recipes/libsecret/all/conanfile.py:59]: We should also delete so there's only less package_id that can be generated?

```suggestion
        if self.settings.os != "Linux":
            del self.options.with_libgcrypt
```

**Review by SSE4 — DISMISSED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**